### PR TITLE
[Rating] getStyles

### LINKED ISSUE #3642: Rating control should have configurable colors.
### Bug Report

- __Package version(s)__: 5.29.0
- __Browser and OS versions__: Chrome, Windows 10

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: (Blocking, High, Normal, Low) Normal

Products/sites affected: (if applicable)

#### Describe the issue:

The rating control stars have fixed front (black) and back colors (grey). I have no way to specify custom colors.
To use colors I need to override using CSS and depend on classnames which may change anytime and break my code.

#### Expected behavior:

The rating control should have provision for specifying custom colors for front and back. These colors should be honored in high contrast mode too.

### PR DESCRIPTION
**Pull request by lambertwang**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #3642
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Converted rating component to getStyles.
2 optional properties added to Rating: getStyles and theme

### CODE REVIEW

File: packages/styling/src/interfaces/ISemanticColors.ts [hunk 1]
```diff
@@ -254,4 +254,21 @@ export interface ISemanticColors {
    * The background color of a checked and hovered list item.
    */
   listItemBackgroundCheckedHovered: string;
+
+  //// Rating
```
Comment (phkuo) [inline packages/styling/src/interfaces/ISemanticColors.ts]: Semantic slots should never be component-specific. They should be usable across a broad range of components.

ratingDefaultColor can use bodySubtext
ratingSelected can use bodyText or bodyTextChecked
ratingDisabled can use disabledBodyText

File: packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    theme
+  } = props;
+
+  const {
+    semanticColors,
+    palette
+  } = theme;
+
+  const ratingSmallIconSize = '16px';
+  const ratingLargeIconSize = '16px';
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts]: These are both the same? Is that intentional?

File: packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+export function getStyles(props: IRatingStyleProps): IRatingStyles {
+  const {
+    disabled,
+    theme
+  } = props;
+
+  const {
+    semanticColors,
+    palette
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts]: are you actually using "palette" here? it should be giving you an unused lint error

File: packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@ export function getStyles(props: IRatingStyleProps): IRatingStyles {
     ratingStarBack: [
       'ms-RatingStar-back',
       {
-        color: semanticColors.bodySubtext,
+        color: semanticColors.disabledText,
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts]: disabled should only be used for non-interactable elements

Comment (lambertwang) [reply packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts]: I need some semantic color that uses the neutralTertiary color. The only other one is inputBorder. Should I make a new semantic color with a lighter tone than bodySubtext, such as bodyHighlightText?

Comment (phkuo) [reply packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts]: no, the point is that we don't have slightly different shades of gray for everything that's of a given concept. e.g. disabled components shouldn't all be different shades of gray, there should be a single  slot in a single place that you can tweak if you wanted all disabled controls to be a faded red.

use bodySubtext, which is the color we use for metadata, descriptions, captions, etc. note that these kinds of things DO contradict some existing redlines... design is usually okay with it if it's only tweaking it by a shade up or down. if design disagrees, point them to me and i'll discuss it with them

Comment (lambertwang) [reply packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts]: For the rating stars, I don't think there is enough contrast between neutralPrimary and neutralSecondary.
star colors

Comment (phkuo) [reply packages/office-ui-fabric-react/src/components/Rating/Rating.styles.ts]: okay yeah that doesn't look good. go ahead and use the palette.neutralTertiary for now for the background

**Comment by Jahnp** [comment]
Adding @betrue-final-final for FYI. From the Screener tests, it looks like the focus rectangle doesn't use the Fabric style:
image

It'd be great to fix that so the test can be accepted. That said, thanks a ton for the changes, @lambertwang!

**Comment by lambertwang** [comment]
This is how the stars look now with the Fabric focus styling. 
fabric focus stars

**Comment by betrue-final-final** [comment]
Perfect!

**Pull request merged**